### DISCUSSION TITLE
add parameter filename to read_json (like read_csv) to include an extra filename column in the result

### DISCUSSION THREAD

**Discussion by erwanj** [duckdb/duckdb#14917] [Ideas]
It would be nice to be able to add an extra column to the result of read_json used to :
- call API with an url or a list of urls : the extra column would contain the url used for the call
- read a json file or a list json files : the extra column would contain the filename

Thus, the behaviour of read_json would be similar to read_csv where we have the parameter :
filename	Whether or not an extra filename column should be included in the result.	BOOL	false

**Comment by Tishj** [comment]
I'm confused, I think this is already possible?
Can you elaborate?
